core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedStageExecution.java

```diff
@@ -395,7 +396,11 @@ private synchronized boolean isStageFlushing()
 
     private synchronized boolean isStageFinished()
     {
-        return finishedTasks.containsAll(allTasks);
+        boolean finished = finishedTasks.size() == allTasks.size();
```

Comment:
If we are 100% sure that `finishedTasks` will not contain any additional tasks which are not included in `allTasks` this is ok, but I would think of `>=` comparison here.

Reply:
> If we are 100% sure that finishedTasks will not contain any additional tasks which are not included in allTasks 

That would be a bug